modules/apps/27-interchain-accounts/controller/keeper/keeper.go

```diff
@@ -85,6 +85,11 @@ func (k Keeper) GetConnectionID(ctx sdk.Context, portID, channelID string) (stri
 	return channel.ConnectionHops[0], nil
 }
 
+// GetChannelKeeper returns the ChannelKeeper
+func (k Keeper) GetChannelKeeper() icatypes.ChannelKeeper {
+	return k.channelKeeper
+}
```

Comment:
this is a bit of a hack, `icatypes.ValidateControllerMetadata` explicitly requires a `ChannelKeeper`, this was added to be able to provide that.

Reply:
hack no longer required